tests/test_adaptive_support.py

```diff
@@ -0,0 +1,133 @@
+            _ = adaptive_support(self.img_src, -0.5)
+
+        with pytest.raises(ValueError, match=r"Given energy_threshold.*"):
+            _ = adaptive_support(self.img_src, 9000)
+
+    def testAdaptiveSupportIncorrectInput(self):
+        """
+        Method should raise meaningful error when passed wrong format input.
+        """
+
+        with pytest.raises(
+            RuntimeError,
+            match="adaptive_support expects `Source` instance or subclass.",
+        ):
+            # Pass numpy array.
+            _ = adaptive_support(np.empty((10, 32, 32)))
+
+    def test_adaptivate_support_hankel(self):
+        """
+        Test support of Hankel function in Real and Fourier space is similar.
+
+        The Fourier transform of `hankel` should be similar to real space,
+        so we can test the support is similar.
+        """
+
+        # Generate stack of Hankel function images.
+        resolution = 64
+        imgs = np.tile(
+            hankel(resolution),
+            (self.n_disc, 1, 1),
+        )
+        # Centered Fourier Transform
+        f_imgs = np.fft.fftshift(np.sqrt(np.abs(np.fft.fft2(imgs))))
+
+        # Setup ImageSource like objects
+        img_src = ArrayImageSource(Image(imgs))
+        f_img_src = ArrayImageSource(Image(f_imgs))
+
+        thresholds = list(self.references.values())
+
+        for threshold in thresholds:
+            _, r = adaptive_support(img_src, threshold)
+            _, rf = adaptive_support(f_img_src, threshold)
+            # Test support is similar between original and transformed
+            self.assertTrue(abs(r - rf) / r < 0.2)
+
+    def test_adaptivate_support_F(self):
+        """
+        Test Fourier support of Gaussian relates to normal distribution.
+        """
+
+        # Generate stack of 2D Gaussian images.
+        imgs = np.tile(
+            gaussian_2d(
+                self.resolution, sigma_x=1 / self.sigma, sigma_y=1 / self.sigma
+            ),
+            (self.n_disc, 1, 1),
+        )
+
+        # Setup ImageSource like objects
+        img_src = ArrayImageSource(Image(imgs))
+
+        thresholds = list(self.references.values())
+
+        for threshold in thresholds:
+            c, _ = adaptive_support(img_src, threshold)
+            # Assert Fourier support is close to normal (doubled for sym).
+            self.assertTrue(abs(2 * c - threshold) / threshold < 0.01)
+
+    def test_adaptivate_support_gaussian_circ(self):
+        """
+        Test against known Gaussian + circ.
+        """
+
+        # for one, two, three, inf standard deviations (one sided)
+        for stddevs, threshold in self.references.items():
+            # Real support should be ~ 1, 2, 3 times sigma
+            c, r = adaptive_support(self.img_src, threshold)
+            # Check closer to this threshold than next
+            logger.info(f"{c} {r} = adaptive_support(..., {threshold})")
+            self.assertTrue(np.abs(r / self.sigma - stddevs) < 0.5)
```

Comment:
It's not clear to me where the disks come in here?

Reply:
Discs are in `self.img_src`.  They are added so that gaussian has some variation.  Otherwise, you are right, in this iteration they are not used in any fundamental way.  

There was a point when I tried using them but they didn't really work well enough for unit testing and I decided to try using the hankel/inverse_r function instead.  Seemed more useful.

The `circ` function is left in mainly for future convenience.

Reply:
Right, but don't we expect the disks to alter the estimated support in the Fourier domain? They decay much slower than the Gaussian.

Reply:
They do, but I had trouble quantifying the Fourier radius result from this method in any meaningful way for a unit test.  Maybe with the change in 03fa86a it's better ... but that is the fourth root of the original formula 😱 ...

Reply:
Ok then I suggest we get rid of them. It's not obvious how the thresholds of the Gaussian and the disk interact. We could just have a separate test with the disks, but if that's not working, we can just skip it.

Reply:
With the reversion in https://github.com/ComputationalCryoEM/ASPIRE-Python/pull/432/commits/b72b3c7c146b0610090b31290aaaf7ef934969d4 , we're not going to get any meaningful output here afaict.

I would like to leave the discs so the code is at least being smoke tested, if that's alright.